RELEASE/scripts/autoscend/combat/auto_combat_default_stage2.ash

```diff
@@ -214,18 +214,70 @@ string auto_combatDefaultStage2(int round, monster enemy, string text)
 	}
 	
 	# Instakill handler
-	boolean doInstaKill = true;
-	if($monsters[Lobsterfrogman, Ninja Snowman Assassin] contains enemy)
+	boolean couldInstaKill = true;
+	if($monsters[Smut Orc Pipelayer,Smut Orc Jacker,Smut Orc Screwer,Smut Orc Nailer] contains enemy && get_property("chasmBridgeProgress").to_int() < 30)
+	{
+		//want to do cold damage in stage3
+		if(my_adventures() > 6)
+		{
+			couldInstaKill = false;
+		}
+	}
+	else if($monsters[Lobsterfrogman, Ninja Snowman Assassin] contains enemy)
 	{
 		if(auto_have_skill($skill[Digitize]) && (get_property("_sourceTerminalDigitizeMonster") != enemy))
 		{
-			doInstaKill = false;
+			couldInstaKill = false;
 		}
 	}
 
-	if(instakillable(enemy) && !isFreeMonster(enemy) && doInstaKill)
+	if(instakillable(enemy) && !isFreeMonster(enemy) && couldInstaKill)
 	{
-		if(canUse($skill[lightning strike]))
+		boolean wantInstaKill;
+		
+		boolean waitForDesert;	//free kills can save turns of Ultrahydrated
+		if(get_property("desertExploration").to_int() < 100 && !isActuallyEd())	//need to explore desert
+		{
+			int currentDesertProgressPerTurn = 1 + 
+			(get_property("bondDesert").to_boolean() ? 2 : 0) +
+			(get_property("peteMotorbikeHeadlight") == "Blacklight Bulb" ? 2 : 0) +
+			(my_familiar() == $familiar[Melodramedary] ? 1 : 0) +
+			equipped_amount($item[survival knife]) +
```

Comment:
Spading shows that knife adds 2 and doesn't stack if more than 1 is equipped. There was a change to how this worked a few days after may day iotm came out